src/components/data-exploration/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -29,14 +30,10 @@ const DiffExprCompute = (props) => {
   const { properties, hierarchy } = useSelector(getCellSets());
   const [isFormValid, setIsFormValid] = useState(false);
   const [numSamples, setNumSamples] = useState(1);
+  const [sampleKeys, setSampleKeys] = useState([])
   const comparisonGroup = useSelector((state) => state.differentialExpression.comparison.group);
   const selectedComparison = useSelector((state) => state.differentialExpression.comparison.type);
-
-
-  // Metadata marks whether cells belong to the same sample/group
-  // Therefore `between samples/groups` analysis makes no sense if there is no metadata.
-  // In the base state, assume there is no metadata. A check for this is done in a useEffect block below.
-  const [hasMetadata, setHasMetadata] = useState(false);
```

Comment:
I remove this check because it is not relevant anymore. We only disable the `within` comparison if there is only 1 sample. If there is more than 1 sample, but no metadata, we can always compare between samples, making this check useless. I've verified this with @alexvpickering